src/pyfmi/fmi_algorithm_drivers.py

```diff
@@ -351,7 +351,9 @@ def __init__(self,
             event_info = self.model.get_event_info()
             if event_info.upcomingTimeEvent and event_info.nextEventTime == model.time:
                 self.model.event_update()
-        
+        elif isinstance(self.model, fmi.FMUModelME2):
+            self.model.event_update()

```

Comment:
what about the event_info condition ?
I see fmi2 also has event_info but upcomingTimeEvent is named nextEventTimeDefined

Reply:
Thanks @jschueller!  And really sorry for my delay here.  But after some further testing compiling my test fmu (from https://github.com/modelon-community/PyFMI/issues/72) as an ME1 instead of ME2, it seems the ``if`` condition in Line 352 is also insufficient to detect an event.  I'm finding that ``event_info.upcomingTimeEvent`` always returns false and ``event_info.nextEventTime`` always returns 0.0.  Similarly, if applied to ME2, ``event_info.nextEventTimeDefined`` always returns false and ``event_info.nextEventTime`` always returns 0.0.  Therefore, in order to get my expected performance, the ``self.model.event_update()`` needs to be called as I suggest, without the ``if``condition on ``event_info``, in both the ME1 and ME2 cases.  I'm still not sure if I'm missing something here though.